brian2/equations/equations.py

```diff
@@ -251,13 +264,38 @@ def dimensions_and_type_from_string(unit_string):
     if unit_string == 'integer':
         return DIMENSIONLESS, INTEGER
 
-    # Check first whether the expression evaluates at all, using only base units
+    # Check first whether the expression only refers to base units
```

Comment:
Actually, this certainly belongs here so maybe the previous bit should stay here too.